doc/service-tests.md

```diff
@@ -7,255 +7,249 @@ automated tests should be included. They serve three purposes:
-use the more concise `expectJSON()` in place of `expectJSONTypes()`.
+In this case we are expecting an object literal instead of a pattern so we should use `expectJSON()` instead of `expectJSONTypes()`. This is more concise and gives us a more helpful error message if the test fails.
 
-Next, we want to cover the code in the `catch` block. To do this, we need to
-trigger an exception. After studying the code, we realize this could happen on
-a request without a Content-Disposition header.
+We also want to include a test for the 'private application not supported' case. One way to do this would be to find another example of a private project which is unlikely to change. For example:
 
-Since we don't have an easy way to get the server to return a real repository
-request without a Content-Disposition header, we will intercept the request
-and provide our own mock response. We use the `intercept()` method provided by
-the [icedfrisby-nock plugin][icedfrisby-nock]. It takes a setup function,
-which returns an interceptor, and exposes the full API of the HTTP mocking
-library [Nock][].
+```js
+t.create('Build status (private application)')
+  .get('/build/wercker/blueprint.json')
+  .expectJSON({ name: 'build', value: 'private application not supported' })
+```
+
+## (5) Mocking Responses
+
+If we didn't have a stable example of a private project, another approach would be to mock the response. An alternative test for the 'private application' case might look like:
 
 ```js
-t.create('missing content-disposition header')
-  .get('/foo/bar.json')
-  .intercept(nock => nock('https://api.travis-ci.org')
-    .head('/foo/bar.svg')
-    .reply(200))
-  .expectJSON({ name: 'build', value: 'invalid' });
+t.create('Build status (private application)')
+  .get('/build/wercker/go-wercker-api.json')
+  .intercept(nock => nock('https://app.wercker.com/api/v3/applications/')
+    .get('/wercker/go-wercker-api/builds?limit=1')
+    .reply(401)
+  )
+  .expectJSON({ name: 'build', value: 'private application not supported' })
 ```
 
+This will intercept the request and provide our own mock response.
+We use the `intercept()` method provided by the
+[icedfrisby-nock plugin][icedfrisby-nock]. It takes a setup function,
+which returns an interceptor, and exposes the full API of the HTTP mocking
+library [Nock][].
+
 Nock is fussy. All parts of a request must match perfectly for the mock to
-take effect, including the method (in this case HEAD), scheme (https), host,
+take effect, including the HTTP method (in this case GET), scheme (https), host,
 and path.
 
 [icedfrisby-nock]: https://github.com/paulmelnikow/icedfrisby-nock#usage
 [Nock]: https://github.com/node-nock/nock
 
+Sometimes it may be also helpful to include service tests which receive a known value from the API. For example, in the `render()` method of our service, there is some logic which sets the badge color based on the build status:
```

Comment:
In my opinion every service should have a test checking a specific value and in most cases it can be achieved by mocking a response. Test validators are checking if value has one of allowed values and I can image a build badge which returns always `failing` or a download badge which returns always `1` and all tests are passing.

Reply:
I can see you addressed this issue. Thanks!